File: python/Roman to Integer.py
class Solution(object):
    def romanToInt(self, s):
        ROMAN_MAP = {
            'M': 1000,
            'D': 500,
            'C': 100,
            'L': 50,
            'X': 10,
            'V': 5,
            'I': 1
        }

        total = 0
        prev = 0
        for i in range(len(s)):
            curr = ROMAN_MAP[s[i]]
            if curr > prev:
                total += curr - 2 * prev
            else:
                total += curr
            prev = curr
        return total

File: cpp/Roman to Integer.cpp
class Solution {
public:
    int romanToInt(string s) {
        std::map<char, int> translate {{'I', 1}, {'V', 5}, {'X', 10}, {'L', 50}, {'C', 100}, {'D', 500}, {'M', 1000}};
        int lastValue = 0;
        int total = 0;
        for (auto digit : s) {
            auto currentValue = translate[digit];
            if (lastValue < translate[digit]) {
                total += currentValue - 2*lastValue;
            } else {
                total += currentValue;
            }

            lastValue = currentValue;
        }

        return total;
    }
};
